core/src/main/java/org/elasticsearch/search/lookup/LeafDocLookup.java

```diff
@@ -68,13 +68,11 @@ public void setDocument(int docId) {
 
     @Override
     public ScriptDocValues<?> get(Object key) {
-        // assume its a string...
-        String fieldName = key.toString();
-        ScriptDocValues<?> scriptValues = localCacheFieldData.get(fieldName);
+        ScriptDocValues<?> scriptValues = localCacheFieldData.get(key);
         if (scriptValues == null) {
-            final MappedFieldType fieldType = mapperService.fullName(fieldName);
+            final MappedFieldType fieldType = mapperService.fullName((String) key);
```

Comment:
Should we rather do `if (key instanceof String == false) { return null; }` to avoid class cast exceptions?